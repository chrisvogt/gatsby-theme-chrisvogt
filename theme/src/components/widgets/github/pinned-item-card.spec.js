import React from 'react'
import renderer from 'react-test-renderer'

import PinnedItemCard from './pinned-item-card'

describe('Widget/GitHub/PinnedItemCard', () => {
  describe('snapshots', () => {
    it('matches the placeholder snapshot', () => {
      const tree = renderer.create(<PinnedItemCard type='placeholder' />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('matches the repository variant snapshot', () => {
      const mockRepositoryItem = {
        description: 'A fake NodeJS project.',
        nameWithOwner: 'themeuser/sample-repo',
        openGraphImageUrl: './fake-image-path.png',
        updatedAt: '1592808981'
      }
      const tree = renderer.create(<PinnedItemCard item={mockRepositoryItem} type='Repository' />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
