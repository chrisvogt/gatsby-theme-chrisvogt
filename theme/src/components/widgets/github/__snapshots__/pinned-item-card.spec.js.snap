// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget/GitHub/PinnedItemCard snapshots matches the placeholder snapshot 1`] = `
<div
  className="css-1yz7e9k"
>
  <div
    className="show-loading-animation"
  >
    <div
      className="css-z5fbhm"
    >
      <div>
        <div
          className="rect-shape"
          style={
            {
              "backgroundColor": "#efefef",
              "height": 40,
              "marginBottom": "2em",
              "marginRight": 10,
              "width": 40,
            }
          }
        />
      </div>
      <div
        className="css-fjakwv"
      >
        <div
          className="text-block"
          style={
            {
              "width": "100%",
            }
          }
        >
          <div
            className="text-row"
            style={
              {
                "backgroundColor": "#efefef",
                "height": "1em",
                "marginTop": 0,
                "maxHeight": "33.333333333333336%",
                "width": "97%",
              }
            }
          />
          <div
            className="text-row"
            style={
              {
                "backgroundColor": "#efefef",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "33.333333333333336%",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="text-block"
      style={
        {
          "width": "100%",
        }
      }
    >
      <div
        className="text-row"
        style={
          {
            "backgroundColor": "#efefef",
            "height": "1em",
            "marginTop": 0,
            "maxHeight": "20%",
            "width": "97%",
          }
        }
      />
      <div
        className="text-row"
        style={
          {
            "backgroundColor": "#efefef",
            "height": "1em",
            "marginTop": "0.7em",
            "maxHeight": "20%",
            "width": "100%",
          }
        }
      />
      <div
        className="text-row"
        style={
          {
            "backgroundColor": "#efefef",
            "height": "1em",
            "marginTop": "0.7em",
            "maxHeight": "20%",
            "width": "94%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Widget/GitHub/PinnedItemCard snapshots matches the repository variant snapshot 1`] = `
<div
  className="css-1yz7e9k"
>
  <div
    className="css-1kbinjp"
  >
    <h4
      className="css-cpvk2h"
    >
      chrisvogt/null
    </h4>
    <span
      className="css-gulexy"
    >
      A fake NodeJS project.
    </span>
    <div
      className="css-1c6hy0j"
    >
      <span>
        Last updated 
        NaN years ago
      </span>
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-up-right-from-square fa-xs "
          data-icon="up-right-from-square"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M352 0c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9L370.7 96 201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L416 141.3l41.4 41.4c9.2 9.2 22.9 11.9 34.9 6.9s19.8-16.6 19.8-29.6l0-128c0-17.7-14.3-32-32-32L352 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
            fill="currentColor"
            style={{}}
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
